api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
Well having the timeout changes the behavior at the moment and will print what can be a scary warning/error to people that got used to the old behavior. If we add a "detach" event in another PR, then we will be able to print with certainty that the container was detached before returning.

Detaching from the tty shouldn't affect the event stream, so we should get the "die" event once the container exits.